addons/pos_pinelabs/models/pos_payment.py

```diff
@@ -0,0 +1,7 @@
+from odoo import models, fields
+
+class PosPayment(models.Model):
+
+    _inherit = "pos.payment"
+
+    plutus_transaction_reference = fields.Char(string='Pine Labs PlutusTransactionReferenceID', help='Required during the refund order process.')
```

Comment:
i know that i suggested creating this field, but with the current implementation i don't see it's use. The way i was thinking was that we would sync the order with the backend before starting the payment process. That way we wouldn't need to keep this transaction reference in the local storage.

I would suggest either to sync the order from the start, or to remove this field and indeed store the transaction reference in the browser, but not in local storage, but rather in the `uiState` property of the payment record. That way it should be automatically synced in the indexedDB

Reply:
I initially thought we should store the `PlutusTransactionReferenceID` only after the payment is completed. While the payment is pending, we could access it from local storage. However, if it works better for you, we can store the `PlutusTransactionReferenceID` in the field(`plutus_transaction_reference`) as soon as the payment request is initiated, rather than in uiState. This way, we can easily access its value directly from the field whenever needed.

Does this approach work for you?
Should I store the `payment_ref_no` before the payment process begins?